myadmin-service-api/src/main/java/za/ac/unisa/myadmin/generic/services/GenericService.java

```diff
@@ -0,0 +1,20 @@
+package za.ac.unisa.myadmin.generic.services;
+
+import za.ac.unisa.myadmin.common.exceptions.DoesNotExistException;
+import za.ac.unisa.myadmin.common.exceptions.InvalidParameterException;
+import za.ac.unisa.myadmin.common.exceptions.MissingParameterException;
+import za.ac.unisa.myadmin.common.exceptions.OperationFailedException;
+import za.ac.unisa.myadmin.generic.dto.GenericCodeInfo;
+import za.ac.unisa.myadmin.generic.dto.GenericMessageInfo;
+
+import java.util.List;
+
+/**
+ * Created by Adrian on 2018-07-27.
+ */
+public interface GenericService {
```

Comment:
Since both said "Gen" thought might be commonly used together elsewhere and probably didn't want to split if it's 1 call each.
Yeah GenericService at this stage just isn't very descriptive... 
A split as suggested probably would make the intention clearer.
Will go with splitting if there isn't any other thoughts here.